lms/djangoapps/email_marketing/tests/test_signals.py

```diff
@@ -523,3 +525,101 @@ def test_modify_email(self, mock_update_user):
+        self.error = error
+        self.code = code
+
+    def is_ok(self):
+        """
+        Return true of no error
+        """
+        return self.error is None
+
+    def get_error(self):
+        """
+        Get error description
+        """
+        return MockSailthruError(self.error, self.code)
+
+
+class MockSailthruError(object):
+    """
+    Mock object for Sailthru Error
+    """
+
+    def __init__(self, error, code=1):
+        self.error = error
+        self.code = code
+
+    def get_message(self):
+        """
+        Get error description
+        """
+        return self.error
+
+    def get_error_code(self):
+        """
+        Get error code
+        """
+        return self.code
+
+
+class SailthruTests(TestCase):
+    """
+    Tests for the Sailthru tasks class.
+    """
+
+    def setUp(self):
+        super(SailthruTests, self).setUp()
+        self.user = UserFactory()
+        self.course_id = CourseKey.from_string('edX/toy/2012_Fall')
+        self.course_url = 'http://lms.testserver.fake/courses/edX/toy/2012_Fall/info'
+        self.course_id2 = 'edX/toy/2016_Fall'
+        self.course_url2 = 'http://lms.testserver.fake/courses/edX/toy/2016_Fall/info'
+
+    @patch('sailthru.sailthru_client.SailthruClient.purchase')
+    @patch('sailthru.sailthru_client.SailthruClient.api_get')
+    @patch('sailthru.sailthru_client.SailthruClient.api_post')
+    def test_update_course_enrollment(self, mock_sailthru_api_post,
+                                      mock_sailthru_api_get, mock_sailthru_purchase):
+        """test update sailthru user record"""
+
+        # create mocked Sailthru API responses
+        mock_sailthru_api_post.return_value = MockSailthruResponse({'ok': True})
+        mock_sailthru_api_get.return_value = MockSailthruResponse({'user': {"id": TEST_EMAIL, "fields": {"vars": 1}}})
+        mock_sailthru_purchase.return_value = MockSailthruResponse({'ok': True})
+        self.user.email = TEST_EMAIL
+        CourseEnrollmentFactory(user=self.user, course_id=self.course_id)
+        with patch('email_marketing.tasks.build_course_url') as m:
+            m.return_value = self.course_url
+            update_course_enrollment(TEST_EMAIL, self.course_id, 'audit')
+        item = [{
+            'url': self.course_url,
+            'price': 0,
+            'qty': 1,
+            'id': 'edX/toy/2012_Fall-audit',
+            'title': 'Course edX/toy/2012_Fall mode: audit'
+        }]
+        mock_sailthru_purchase.assert_called_with(TEST_EMAIL, item, options={})
+
+    @patch('sailthru.sailthru_client.SailthruClient.purchase')
+    def test_verify_dependency_on_flag(self, mock_sailthru_purchase):
+        """
+        make sure sailthru purchase call not invoked in the following conditions:
+            i: waffle switch is 'False'
+            ii: waffle switch is 'True' and course_mode is 'verified'
+        """
+        update_sailthru(None, None, self.user, 'verified', self.course_id)
```

Comment:
I do feel like this line and next belong to a different test case. Can we separate these two lines out into its own test case?